test/e2e/utils/portforwarder/portforwarder.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package portforwarder
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+)
+
+type PortForwarder struct {
+	config        *rest.Config
+	clientset     kubernetes.Interface
+	namespace     string
+	name          string
+	targetPort    int
+	listenAddress string
+	listenPort    int
+	stopCh        chan struct{}
+}
+
+// NewPortForwarder creates Port Forwarder for a Pod / Service
+// After creating Port Forwarder object, call Start() on it to start forwarding
+// channel and Stop() to terminate it
+func NewPortForwarder(
+	config *rest.Config,
+	namespace string,
+	name string,
+	targetPort int,
+	listenAddress string,
+	listenPort int,
+) (*PortForwarder, error) {
+	pf := &PortForwarder{
+		config:        config,
+		namespace:     namespace,
+		name:          name,
+		targetPort:    targetPort,
+		listenAddress: listenAddress,
+		listenPort:    listenPort,
+	}
+
+	var err error
+	pf.clientset, err = kubernetes.NewForConfig(pf.config)
+	if err != nil {
+		return pf, fmt.Errorf("failed to create Kubernetes client: %w", err)
+	}
+
+	return pf, nil
+}
+
+// Start Port Forwarding channel
+func (p *PortForwarder) Start() (int, error) {
+	p.stopCh = make(chan struct{}, 1)
```

Comment:
can we move this to the initializer function?

Reply:
is there an advantage in doing this? what I like with the current version of the code is:
- the same `PortForwarder` object could be reused if needed (not sure how useful it is)
- the code will panic if `Stop` is called without a previous `Start` (I think it's good behavior for test code)

Reply:
Agreed. The current version is better. Just curious why it is initialized in the start function.